Samples/ObservableScreen/Sources/CounterWorkflow.swift

```diff
@@ -0,0 +1,102 @@
+import Foundation
+import Workflow
+import WorkflowSwiftUI
+import WorkflowUI
+
+struct CounterWorkflow: Workflow {
+    // Dependencies from parent.
+    let info: CounterInfo
+    let resetToken: ResetToken
+    let initialValue: Int
+    let maxValue: Int?
+
+    @ObservableState
+    struct State {
+        var count: Int
+        var maxValue: Int?
+
+        var info: CounterInfo
+
+        var boundedCount: Int {
+            if let maxValue {
+                min(count, maxValue)
+            } else {
+                count
+            }
+        }
+    }
+
+    enum Action: WorkflowAction {
+        typealias WorkflowType = CounterWorkflow
+
+        case increment
+        case decrement
+
+        func apply(toState state: inout CounterWorkflow.State) -> CounterWorkflow.Output? {
+            // the max value is applied in the view, not here, so it's possible to increment
+            // above the max
+            switch self {
+            case .increment:
+                state.count += state.info.stepSize
+            case .decrement:
+                state.count -= state.info.stepSize
+            }
+            return nil
+        }
+    }
+
+    typealias Output = Never
```

Comment:
Can you add an example feature where this would be used? Or is the idea that it never should be?